### ISSUE TITLE
Using rails 5.2.1 causes failures

### ISSUE THREAD

**Issue by kbroderick** [pat/joiner#5]
The spec suite fails on eight specs under Rails 5.2.1, all with:

```
     wrong number of arguments (given 4, expected 3)
     # /Users/user/.rvm/gems/ruby-2.5.1@joiner/gems/activerecord-5.2.1/lib/active_record/associations/join_dependency.rb:70:in `initialize'
     # ./lib/joiner/joins.rb:26:in `new'
     # ./lib/joiner/joins.rb:26:in `join_values'
     # ./lib/joiner/joins.rb:22:in `alias_for'
     # ./spec/acceptance/joiner_spec.rb:72:in `block (2 levels) in <top (required)>'
```

This appears to be the result of a change in `activerecord/lib/active_record/associations/join_dependency.rb`, where `initialize` has gone from four parameters to three, in changeset https://github.com/rails/rails/commit/50036e673b3b6ff41476814b212b55fabac3a638

**Comment by pat** [comment]
Appreciate the debugging work and the bug report :) I'll try to get this sorted in the next few days!

**Comment by pat** [comment]
Just released joiner v0.4.1 which has a fix for this.

**Comment by kbroderick** [comment]
Thanks for the quick fix!

We're now happily running joiner 0.4.1 under Rails 5.2.1.
